### PR TITLE
Use spearate testkit dir for each run

### LINKED ISSUE #41431: [CI] Numerous :build-tools:integTest fail in darwin builds
There have been several **darwin** builds where many integration tests of the build tools have failed with little indication of the cause:

* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-darwin-compatibility/350/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+multijob-darwin-compatibility/124/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+multijob-darwin-compatibility/125/console

I tried to reproduce this locally on my mac using `./gradlew :build-tools:integTest` on both the 7.0 and master branches but that command was successful for both branches.

The errors in the build log are not very expansive about what went wrong:

```
org.elasticsearch.gradle.precommit.ThirdPartyAuditTaskIT > testViolationFoundAndCompileOnlyIgnored FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at ThirdPartyAuditTaskIT.java:32

org.elasticsearch.gradle.precommit.ThirdPartyAuditTaskIT > testWithEmptyRules FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at ThirdPartyAuditTaskIT.java:32

org.elasticsearch.gradle.precommit.ThirdPartyAuditTaskIT > testJarHellWithJDK FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at ThirdPartyAuditTaskIT.java:32

org.elasticsearch.gradle.precommit.ThirdPartyAuditTaskIT > testClassNotFoundAndCompileOnlyIgnored FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at ThirdPartyAuditTaskIT.java:32

org.elasticsearch.gradle.precommit.JarHellTaskIT > testJarHellDetected FAILED
    java.lang.AssertionError at JarHellTaskIT.java:34

org.elasticsearch.gradle.precommit.ThirdPartyAuditTaskIT > testElasticsearchIgnoredWithViolations FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at ThirdPartyAuditTaskIT.java:32

org.elasticsearch.gradle.precommit.ThirdPartyAuditTaskIT > testElasticsearchIgnored FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at ThirdPartyAuditTaskIT.java:32

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testNamingConvention FAILED
    java.lang.AssertionError at TestingConventionsTasksIT.java:50

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testValidSetupWithBaseClass FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at TestingConventionsTasksIT.java:103

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testValidSetupWithoutBaseClass FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure at TestingConventionsTasksIT.java:96

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testTaskNotImplementBaseClass FAILED
    java.lang.AssertionError at TestingConventionsTasksIT.java:83

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testNoEmptyTasks FAILED
    java.lang.AssertionError at TestingConventionsTasksIT.java:63

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testTestsInMain FAILED
    java.lang.AssertionError at TestingConventionsTasksIT.java:111

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testAllTestTasksIncluded FAILED
    java.lang.AssertionError at TestingConventionsTasksIT.java:73

org.elasticsearch.gradle.precommit.TestingConventionsTasksIT > testInnerClasses FAILED
    java.lang.AssertionError at TestingConventionsTasksIT.java:36
```

For example `ThirdPartyAuditTaskIT.java:32` is the `.build()` line in:

```
        // Build the sample jars
        getGradleRunner("thirdPartyAudit")
            .withArguments("build", "-s")
            .build();
```

### PR DESCRIPTION
**Pull request by alpar-t**
Gradle Testkit reuses the teskit dir by default between tests.
With this change we use a temporary one for each run
hoping it will fix #41431

### CODE REVIEW

File: buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java [hunk 1]
```diff
@@ -86,6 +86,7 @@ public void testCurrentExamplePlugin() throws IOException {
 
         GradleRunner.create()
             .withProjectDir(tmpDir.getRoot())
+            .withTestKitDir(tmpDir.newFolder("testkit"))
```
Comment (mark-vieira) [inline buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java]: This shouldn't be necessary. We are already running each of these test cases in a new temporary directory. By default TestKit is going to create a `.gradle-test-kit` directory under that folder. Really all this is doing is renaming that folder `testkit` instead. If the example plugin tests are failing, it's not due to testkit directory reuse.

File: buildSrc/src/test/java/org/elasticsearch/gradle/testclusters/TestClustersPluginIT.java [hunk 2]
```diff
@@ -21,11 +21,19 @@
 import org.elasticsearch.gradle.test.GradleIntegrationTestCase;
 import org.gradle.testkit.runner.BuildResult;
 import org.gradle.testkit.runner.GradleRunner;
+import org.junit.Before;
 
 import java.util.Arrays;
 
 public class TestClustersPluginIT extends GradleIntegrationTestCase {
 
+    private GradleRunner testclusters;
```
Comment (mark-vieira) [inline buildSrc/src/test/java/org/elasticsearch/gradle/testclusters/TestClustersPluginIT.java]: Can we rename this something more appropriate, like `runner`?

**Review by mark-vieira — APPROVED** [review]
Couple of comments. Otherwise LGTM 👍

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**